server/consumer.go

```diff
@@ -1586,6 +1586,27 @@ func (o *consumer) processAckMsg(sseq, dseq, dc uint64, doSample bool) {
 	}
 }
 
+// Determine if this is a truly filtered consumer. Modern clients will place filtered subjects
+// even if the stream only has a single non-wildcard subject designation.
+// Read lock should be held.
+func (o *consumer) isFiltered() bool {
+	if o.cfg.FilterSubject == _EMPTY_ {
+		return false
+	}
+	// If we are here we want to check if the filtered subject is
+	// a direct match for our only listed subject.
+	mset := o.mset
+	if mset == nil {
+		return true
```

Comment:
is `mset == nil` a normal condition? If so, what does that mean? (so I understand why we would return `true` here).

Reply:
We remove the mset from the consumer when deleted. Will change that up but not in this PR, so its there for safety.

Reply:
My question is then: when that is the case, does it makes more sense to return `true` (is filtered) or `false`? Or does it not matter..

Reply:
In this case we do not want to assume it is not.